test/studies/parboil/histo/histoSerial.chpl

```diff
@@ -102,7 +102,7 @@ proc dumpHisto(histo: [] uint(8), height: uint(32), width: uint(32), outFilename
       if value == 0 then
         (pixMap[pos].R, pixMap[pos].G, pixMap[pos].B) = (0:uint(8),0:uint(8),0:uint(8));
       else
-        pixMap[pos] = HSVtoRGB(0.0, 1.0, cbrt(1+63.0*value/max(uint(8)))/4);
+        pixMap[pos] = HSVtoRGB(0.0, 1.0, cbrt(1+63.0*value:real/max(uint(8)))/4);
```

Comment:
Here, `value` is a `uint(8)`. Since everything else in sight in `cbrt(1+63.0*value:real/max(uint(8)))/4)` is `param`, the type of `value` determines the result type and we get a `real(32)` out. I think this, in itself, is arguably surprising and probably not what was intended when this code was written. (And, at the time it was written, this expression would have been `real` rather than `real(32)`).

> This case feels similar to (and similarly unfortunate as) int8ToReal32 in that, presumably, even if we added a real(16) or real(8), this would be legal and end up doing a real(64) computation?

It could be `real(8)` or `real(16)`, because everything else is param.

> Remind me, which usually wins given the choice between a param downcast and an implicit conversion?

We use param downcast / narrowing as the last tie-breaker [1](https://chapel-lang.org/docs/language/spec/procedures.html#determining-most-specific-functions). So if we had decided already that some candidate was "better" for some other reason, the param downcast part won't come in to play. 

> This case makes me head hurt. Which maybe suggests that the :real is necessary because of the param. But if 63.0 were myReal64, I don't think it should be.

That's also my understanding: the `:real` would not be necessary if it included a `myReal64` in the computation. In fact, that would make the warning not fire, because it won't warn for converting something to `real` aka `real(64)` since that isn't surprising.